src/Eloquent/Graph/Collection.php

```diff
@@ -16,7 +16,8 @@ class Collection extends Base
      * Generate a nested tree.
      *
      * @param string $childrenRelation
-     * @return $this<int, TModel>
+     * @return static<TKey, TModel>
+     * @phpstan-ignore return.phpDocType
```

Comment:
I like to always have a comment on why something is ignored (helps for other readers / when reading later)

```suggestion
     * @phpstan-ignore return.phpDocType (...)
```